### PR TITLE
chore(agoric-cli): Type-check *.js files

### PR DESCRIPTION
**Pull request by gibson042**
## Description
Includes agoric-cli *.js files in type checking.

### Security Considerations
n/a

### Scaling Considerations
n/a

### Documentation Considerations
n/a

### Testing Considerations
n/a

### Upgrade Considerations
None known.

### CODE REVIEW

File: packages/agoric-cli/src/chain-config.js [hunk 1]
```diff
@@ -92,6 +97,8 @@ export function finishCosmosApp({
   rosettaPort = `${DEFAULT_ROSETTA_PORT}`,
 }) {
   const rpcPort = Number(portNum);
+  // TODO: Use an accurate narrow type.
+  /** @type {any} */
```
Comment (michaelfig) [inline packages/agoric-cli/src/chain-config.js]: We can at least narrow a tiny bit based on the TOML library we use:
```suggestion
  /** @type {Record<string, any>} */
```

File: packages/agoric-cli/src/chain-config.js [hunk 2]
```diff
@@ -151,6 +163,8 @@ export function finishTendermintConfig({
   const rpcPort = Number(portNum);
 
   // Adjust the config.toml.
+  // TODO: Use an accurate narrow type.
+  /** @type {any} */
```
Comment (michaelfig) [inline packages/agoric-cli/src/chain-config.js]:
```suggestion
  /** @type {Record<string, any>} */
```

File: packages/agoric-cli/src/cosmos.js [hunk 3]
```diff
@@ -50,7 +50,8 @@ export default async function cosmosMain(progname, rawArgs, powers, opts) {
                 },
               );
               // Ensure the build doesn't mess up stdout.
-              ps.childProcess.stdout.pipe(process.stderr);
+              // @ts-expect-error stdout won't be null
+              ps.childProcess?.stdout.pipe(process.stderr);
```
Comment (michaelfig) [inline packages/agoric-cli/src/cosmos.js]: Was this a mistaken optional chaining of childProcess?  Maybe you meant:
```suggestion
              ps.childProcess.stdout?.pipe(process.stderr);
```

File: packages/agoric-cli/src/install.js [hunk 4]
```diff
@@ -37,7 +37,8 @@ export default async function installMain(progname, rawArgs, powers, opts) {
       stdio: ['inherit', 'pipe', 'inherit'],
     });
     const stdout = [];
-    p.childProcess.stdout.on('data', out => stdout.push(out));
+    // @ts-expect-error stdout won't be null
+    p.childProcess?.stdout.on('data', out => stdout.push(out));
```
Comment (michaelfig) [inline packages/agoric-cli/src/install.js]:
```suggestion
    p.childProcess.stdout?.on('data', out => stdout.push(out));
```

File: packages/agoric-cli/src/install.js [hunk 5]
```diff
@@ -169,8 +170,8 @@ export default async function installMain(progname, rawArgs, powers, opts) {
       )
         .then(results => {
           // After all have settled, throw any errors.
-          const failures = results.filter(
-            ({ status }) => status !== 'fulfilled',
+          const failures = /** @type {PromiseRejectedResult[]} */ (
+            results.filter(({ status }) => status !== 'fulfilled')
           );
```
Comment (michaelfig) [inline packages/agoric-cli/src/install.js]: No cast necessary, but it only narrows if you don't destructure the status from the result:

```suggestion
          const failures = results.filter(result => result.status !== 'fulfilled');
```

File: packages/agoric-cli/src/install.js [hunk 6]
```diff
@@ -291,6 +292,7 @@ export default async function installMain(progname, rawArgs, powers, opts) {
         await pspawn('yarn', [...linkFlags, 'unlink', pjName]);
         return pspawn('yarn', [...linkFlags, 'link'], {
           stdio: 'inherit',
+          // @ts-expect-error if dir is null
```
Comment (michaelfig) [inline packages/agoric-cli/src/install.js]: Could you do something like the following at the top of this function, then avoid these `@ts-expect-error`s?

```js
  await Promise.all(
    [...sdkPackageToPath.entries()].map(async ([pjName, dir]) => {
      if (dir == null) {
        // We don't expect a nullish `dir`.
        return;
      }
 ```

**Review by michaelfig — COMMENTED** [review]
Great job!  I just have some clarifications to avoid typecasts and `@ts-expect-error` where practical.

**Comment by gibson042** [comment]
Thanks @michaelfig; all suggestions accepted.

**Review by michaelfig — APPROVED** [review]
Fantastic!  :shipit:

**Review by turadg — APPROVED** [review]

**Pull request merged**